### DISCUSSION TITLE
Valhalla road network graph tiles from GCS bucket

### DISCUSSION THREAD

**Discussion by girishyadav20** [valhalla/valhalla#4847] [Q&A] [answered]
Hi, 

I have stored the valhalla graph tile(.gph) files in a Google Cloud Storage bucket in the proper directory structure /<level>/<tile_id>/<number>.gph. What are the options to assign the bucket as the "tile_dir" configuration? 
- If I specify the GCS bucket object url (https://cloud.google.com/storage/docs/request-endpoints#typical) in "tile_url" configuration, how the authentication would be handled? Is there any configuration to provide auth token?
- How does the "tile_url" works? Does it downloads the required tile files locally in "tile_dir" location?

Thanks!

**Comment by kevinkreiser** [comment, marked as answer]
1. we do not provide an implementation for authentication of any specific service in valhalla itself. a typical use case for `tile_url` will involve a proxy that knows how to do whatever auth you need to get tiles out of gcs or s3 or whatever.
2. yes if you have the `tile_url` configured properly the software will pull tiles from this endpoint on the fly and yes it will put them on disk locally if your `tile_dir` is configured

**Reply by girishyadav20** [reply]
Thanks @kevinkreiser
